src/View/StringTemplate.php

```diff
@@ -323,4 +324,68 @@ protected function _formatAttribute($key, $value, $escape = true)
 
         return $key . '="' . ($escape ? h($value) : $value) . '"';
     }
+
+    /**
+     * Adds a class and returns a unique list either in array or space separated
+     *
+     * ### Options
+     *
+     * - `useIndex` if you are inputting an array with an 'element' other than 'class'.
+     *              Also setting to 'false' will manipulate the whole array (default is 'class')
+     * - `asString` Setting this to `true` will return a space separated string (default is `false`)
```

Comment:
This method comes from a Utility class I have (slightly modified). Ideally this method would live in HtmlHelper or a Html Utility class like `Cake\Utility\Text` so it can be used by end-user. Preferable a static Utility class then its easier to use in Core. With this in mind I think `asString` would be useful. However I'm happy to remove the option in this context.